src/compas_viewer/scene/geometryobject.py

```diff
@@ -0,0 +1,124 @@
+from typing import Optional
+from typing import Tuple
+
+from compas.colors import Color
+from compas.geometry import Geometry
+from compas.geometry import Line
+from compas.geometry import Point
+from compas.scene import GeometryObject as BaseGeometryObject
+
+from .sceneobject import ShaderDataType
+from .sceneobject import ViewerSceneObject
+
+
+class GeometryObject(ViewerSceneObject, BaseGeometryObject):
+    """Viewer scene object for displaying COMPAS Geometry.
+
+    Parameters
+    ----------
+    geometry : :class:`compas.geometry.Geometry`
+        A COMPAS geometry.
+    pointcolor : :class:`compas.colors.Color`, optional
+        The color of the points. Global settings in the viewer will be used if not specified.
+    linecolor : :class:`compas.colors.Color`, optional
+        The color of the lines. Global settings in the viewer will be used if not specified.
+    surfacecolor : :class:`compas.colors.Color`, optional
+        The color of the surfaces. Global settings in the viewer will be used if not specified.
+    **kwargs : dict, optional
+        Additional options for the :class:`compas_viewer.scene.ViewerSceneObject`
+        and :class:`compas.scene.GeometryObject`.
+
+    Attributes
+    ----------
+    geometry : :class:`compas.geometry.Geometry`
+        The COMPAS geometry.
+    pointcolor : :class:`compas.colors.Color`
+        The color of the points.
+    linecolor : :class:`compas.colors.Color`
+        The color of the lines.
+    surfacecolor : :class:`compas.colors.Color`
+        The color of the surfaces.
+    mesh : :class:`compas.datastructures.Mesh`
+        The triangulated mesh representation of the geometry.
+    LINEARDEFLECTION : float
+        The default linear deflection for the geometry.
+
+    See Also
+    --------
+    :class:`compas.geometry.Geometry`
+    """
+
+    LINEARDEFLECTION = 0.2
+
+    def __init__(
+        self,
+        geometry: Geometry,
+        pointcolor: Optional[Color] = None,
+        linecolor: Optional[Color] = None,
+        surfacecolor: Optional[Color] = None,
+        **kwargs
+    ):
+
+        super().__init__(geometry=geometry, **kwargs)
+        self.geometry: Geometry
+
+        self.pointcolor = pointcolor or self.viewer.config.pointscolor
+        self.linecolor = linecolor or self.viewer.config.linescolor
+        self.surfacecolor = surfacecolor or self.viewer.config.facescolor
+
+    @property
+    def points(self) -> Optional[list[Point]]:
+        """The points to be shown in the viewer."""
+        raise NotImplementedError
+
+    @property
+    def lines(self) -> Optional[list[Line]]:
+        """The lines to be shown in the viewer."""
+        raise NotImplementedError
+
+    @property
+    def surfaces(self) -> Optional[list[Tuple[Point, Point, Point]]]:
+        """The surface to be shown in the viewer. Currently only triangles are supported."""
+        raise NotImplementedError
+
```

Comment:
perhaps it would make sense to add a "viewmesh" property?

Reply:
changed `surfaces` to `viewmesh`

Reply:
that is not really what i meant. many geometry objects use a mesh representation for viewing purposes. currently, it seems that the construction of this "viewmesh" is more or less repeated for many objects. perhaps this coud be generalised with a `viewmesh` property...